app/src/main/java/com/bootcamp_android/parking_app/MainActivity.kt

```diff
@@ -2,25 +2,30 @@ package com.bootcamp_android.parking_app
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
-import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
-import com.bootcamp_android.parking_app.adapter.LotAdapter
+import com.bootcamp_android.parking_app.adapter.Adapter
 import com.bootcamp_android.parking_app.adapter.Provider
 
 class MainActivity : AppCompatActivity() {
 
+    private val dataAdapter: Adapter by lazy {
```

Comment:
check out the difference between lazy and lateinit

Reply:
great i understand with lazy i can use val, but with lateinit i can use  both, any adivice😄?

Reply:
Apart from that, keep in mind that lazy is used when you want to use its value on runtime, when it's needed. The idea is to not initialized variables that are sometimes needed and sometimes not, that's when you should use a lazy.
In that case, you can keep the lazy, no problem, but notice that adapter is always needed in runtime.

Reply:
I have seen that, you are using the adapter in the onCreate method...
I have a rule, if your variable is needed in on create, you shouldn't use lazy with that variable.
Please checkout the lifecycle of an activity.
It's great that you are trying to use lazy, maybe you con use it inside of an onclicklistener (because inside is probably that variable is needed or not)